### PR TITLE
Initial OpenTitan userspace app

### PR DESCRIPTION
**Pull request by osenft**
Note that this version of the code still contains a few "debug" `writeln!` statements with unused `Result` return values. These will be removed / used in a subsequent PR.

```
----------------------
`make prtest` summary:
----------------------
git rev-parse HEAD
db7d6de8a7f8a15ad2a39d1af9dbd4b73b7b83f4
git status
On branch otpilot-base
Your branch is up to date with 'origin/otpilot-base'.

nothing to commit, working tree clean
```

### CODE REVIEW

File: shared-lib/spiutils/src/protocol/flash.rs [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+    pub address: Option<AddrType>,
+}
+
+impl<'a, AddrType> Header<AddrType>
+where AddrType: BeInt {
+    fn from_wire<R: Read<'a>>(mut r: R) -> Result<Self, FromWireError> {
+        let opcode_u8 = r.read_be::<u8>()?;
+        let opcode = OpCode::from_wire_value(opcode_u8).ok_or(FromWireError::OutOfRange)?;
+
+        let mut address : Option<AddrType> = None;
```
Comment (jrvanwhy) [inline shared-lib/spiutils/src/protocol/flash.rs]: Formatting nitpick: remove the space before the `:` character.

File: userspace/otpilot/src/spi_device.rs [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+
+use libtock::result::TockResult;
+use libtock::shared_memory::SharedMemory;
+use libtock::syscalls;
+use libtock::syscalls::raw::yieldk;
+
+pub const MAX_READ_BUFFER_SIZE : usize = 512;
+
+#[allow(dead_code)]
+pub const MAX_WRITE_BUFFER_SIZE : usize = 2048;
```
Comment (jrvanwhy) [inline userspace/otpilot/src/spi_device.rs]: Nitpick: Remove the unnecessary space before the `:` on MAX_READ_BUFFER_SIZE and MAX_WRITE_BUFFER_SIZE.

**Review by jrvanwhy — APPROVED** [review]

**Review by jrvanwhy — CHANGES_REQUESTED** [review]
Oh yeah, please add manticore to the list of submodules cloned by the `ci` workflow so that CI passes. Also, please run `make prtest` and and post the output to confirm that it works on hardware.

File: shared-lib/spiutils/src/protocol/flash.rs (hunk 1 above)
Comment (osenft) [reply shared-lib/spiutils/src/protocol/flash.rs]: Done.

File: userspace/otpilot/src/spi_device.rs (hunk 2 above)
Comment (osenft) [reply userspace/otpilot/src/spi_device.rs]: Done.

**Comment by osenft** [comment]
> Oh yeah, please add manticore to the list of submodules cloned by the `ci` workflow so that CI passes. Also, please run `make prtest` and and post the output to confirm that it works on hardware.

Thank you! This should be all done now.

**Review by jrvanwhy — APPROVED** [review]

**Pull request merged**